wagtail/wagtailadmin/templates/wagtailadmin/block_forms/stream_menu.html

```diff
@@ -1,10 +1,14 @@
 <div class="stream-menu {% if state == 'closed' %}stream-menu-closed{% endif %}" id="{{ menu_id }}">
     <a href="#" class="toggle"><span>Insert block</span></a>
     <div class="stream-menu-inner">
-        <ul>
-            {% for child_block in child_blocks %}
-                <li><button type="button" class="button action-add-block-{{ child_block.name }} icon icon-{{ child_block.meta.icon }}"><span>{{ child_block.label }}</span></button></li>
-            {% endfor %}
-        </ul>
+        {% regroup child_blocks by meta.group as grouped_child_blocks %}
+        {% for child_blocks in grouped_child_blocks %}
+            <h3>{{child_blocks.grouper}}</h3>
```

Comment:
It would be good to add an `{% if child_blocks.grouper %}` around this, so that we don't generate an empty `<h3></h3>` in the default case.